teps/0048-task-results-without-results.md

```diff
@@ -426,6 +434,307 @@ with the git events such as `when a pull request is created`, etc but it
 doesn't runs. Having default results defined in `check-name-matches` Task
 will fix the `Pipeline`.
 
+## Proposal
+
+We propose adding an optional field - `default` - to the `Result` specification in `Task`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task
+spec:
+  results:
+  - name: merge_status
+    description: whether to rebase or squash
+    type: string
+    default: rebase
+  - name: branches
+    description: branches in the repository
+    type: array
+    default:
+    - main
+    - v1alpha1
+    - v1beta1
+    - v1
+  - name: images
+    type: object
+    properties:
+      node:
+       type: string
+       default: "node:latest" 
+      gcloud:
+       type: string
+       default: "gcloud:latest"
+  steps:
+  ...
+```
+
+
+Adding a default value to `Results` is optional; validation of a `Task` doesn't
+fail if the default value isn't provided.
+
+Adding a default value to a `Result` will guarantee that it will hold a value even
+if the `Task` fails to produce it. If a `Task` does not produce a `Result` that does not
+have a default value, then the `Task` should fail - see [tektoncd/pipeline#3497](https://github.com/tektoncd/pipeline/issues/3497) for further details.
+
+The proposed solution can be used to solve the above use cases as follows:
+
+### Consuming `Task` `Result` from the conditional tasks
+
+```yaml
+spec:
+  tasks:
+    # Clone runtime repo
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    # check the content of the PR i.e. the changes proposed
+    # does any of those changes contain changing a dockerfile
+    # if so, build a new image, otherwise, skip building an image
+    - name: check-pr-content
+      runAfter: [ "git-clone" ]
+      taskRef:
+        name: check-pr-content
+      results:
+        - name: image-change
+          default: "no"
```

Comment:
I don't think the `results` can be specified in `pipelineTask`. Is this being introduced here in this proposal in addition to the default specified in `task` specification?

Reply:
@pritidesai actually no 😅. In the use-cases mentioned [above](https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md#consuming-task-results-from-the-conditional-tasks), I saw this syntax, so thought that at the time of writing this TEP the author might have just displayed that the following result would be produced from the `Task`. Happy to change this  😄

Reply:
oh no, I have completely forgotten what I authored 😞 

yeah if I can recollect, it's included just for the readability 😥 

let's drop `results` from the `pipelineTask` since its not something supported or will be as part of this proposal, otherwise looks good to me.